File: 1493-longest-subarray-1s-after-deleting-1-ele.py
# similar to 1004, k is always 1 though... solution needs - 1 for some reason too...

class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        k = 1
        start = 0
        end = 0

        while end < len(nums):
            if nums[end] == 0:
                k -= 1

            if k < 0:
                if nums[start] == 0:
                    k += 1
                start += 1

            end += 1

        return end - start - 1

File: 1493-longest-subarray-1s-after-deleting-1-ele.js
// see py file for thought process

var longestSubarray = function(nums) {
    let k = 1;
    let start = 0;
    let end = 0;

    while (end < nums.length) {
        if (nums[end] === 0) k--;

        if (k < 0) {
            if (nums[start] === 0) k++;
            start++;
        }
        end++;
    }

    return end - start - 1;
};
